### PR TITLE
add methods findById 、findByCondition 、countByCondition 、findOne 、update 、remove

### PR DESCRIPTION
**Pull request by atorber**
add methods findById 、findByCondition 、countByCondition 、findOne 、update 、remove

### CODE REVIEW

**Comment by levg34** [comment]
Thank you for your contribution!

I will have a look when I have some time.
If you feel like it, you may add unit tests in the `test` folder. 

Also, I need to edit the pipeline to run tests for PRs, I have not yet done this on personal projects.

File: src/orm.ts [hunk 1]
```diff
@@ -47,4 +47,40 @@ export class ORM<T extends IID> implements IORM {
...
+
+    static async update<T>(query: Partial<T>, update: Partial<T>, options?: any): Promise<number> {
+        const db = await ORM.getDatabase()
+        const { numAffected } = await db.updateAsync(query, { $set: update }, options)
+        return numAffected
+    }
+
+    static async remove<T>(query: Partial<T>, options?: any): Promise<number> {
+        const db = await ORM.getDatabase()
+        const { numRemoved } = await db.removeAsync(query, options)
```
Comment (levg34) [inline src/orm.ts]:
```suggestion
        const numRemoved = await db.removeAsync(query, options)
```

File: src/orm.ts [hunk 2]
```diff
@@ -47,4 +47,40 @@ export class ORM<T extends IID> implements IORM {
         const docs = await db.findAsync(query)
         return docs
     }
+
+    static async findOne<T>(query: Partial<T>): Promise<T | null> {
+        const db = await ORM.getDatabase()
+        const doc = await db.findOneAsync(query)
+        return doc
+    }
```
Comment (levg34) [inline src/orm.ts:55]: 👍🏻

File: src/orm.ts [hunk 3]
```diff
@@ -47,4 +47,40 @@ export class ORM<T extends IID> implements IORM {
...
+
+    static async update<T>(query: Partial<T>, update: Partial<T>, options?: any): Promise<number> {
+        const db = await ORM.getDatabase()
+        const { numAffected } = await db.updateAsync(query, { $set: update }, options)
+        return numAffected
+    }
```
Comment (levg34) [inline src/orm.ts:61]: The save method both creates and updates one specific object, but indeed it cannot be used to do batch updates.

File: src/orm.ts [hunk 4]
```diff
@@ -47,4 +47,40 @@ export class ORM<T extends IID> implements IORM {
...
+        const db = await ORM.getDatabase()
+        const { numRemoved } = await db.removeAsync(query, options)
+        return numRemoved
+    }
+
+    static async findById<T>(id: string): Promise<T | null> {
+        const db = await ORM.getDatabase()
+        const doc = await db.findOneAsync({ _id: id })
+        return doc
+    }
```
Comment (levg34) [inline src/orm.ts:73]: Can be done with `find`, but it is a nice shortcut 👍🏻

File: src/orm.ts [hunk 5]
```diff
@@ -47,4 +47,40 @@ export class ORM<T extends IID> implements IORM {
...
+
+    static async findByCondition<T>(condition: any): Promise<T[]> {
+        const db = await ORM.getDatabase()
+        const docs = await db.findAsync(condition)
+        return docs
+    }
```
Comment (levg34) [inline src/orm.ts]: This is useless, as `find()` already does that. As you can see, the code is identical. I suggest removing this method.

```suggestion
```

File: src/orm.ts [hunk 6]
```diff
@@ -47,4 +47,40 @@ export class ORM<T extends IID> implements IORM {
...
+
+    static async countByCondition<T>(condition: any): Promise<number> {
+        const db = await ORM.getDatabase()
+        const count = await db.countAsync(condition)
+        return count
+    }
```
Comment (levg34) [inline src/orm.ts]: Indeed, there was no way to count before. I suggest renaming it:

```suggestion
    static async count<T>(condition: any): Promise<number> {
        const db = await ORM.getDatabase()
        const count = await db.countAsync(condition)
        return count
    }
```

**Review by levg34 — CHANGES_REQUESTED** [review]
Great job!

It just lacks some unit tests, and some correction I have suggested.

I do not know how if I can push commits directly to your PR to add things before merging? I have never done this before out of GitHub Enterprise, I do not know how it works on GitHub.com.

Also, it would be nice to update the documentation (the `README.md` file) to reflect the changes brought by this PR.

**Review by atorber — COMMENTED** [review]
👍🏻

File: src/orm.ts (hunk 4 above)
Comment (atorber) [reply src/orm.ts:73]: 👍🏻

**Review by levg34 — APPROVED** [review]
🆗 

Tests and doc missing, but I will add them myself.

**Pull request merged**